### DISCUSSION TITLE
Zone 2 (HR) Training - How to?

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [cagnulein/qdomyos-zwift#1919] [Q&A] [answered]
I see in the app I can set QZ to gradually adjust my bike's resistance to maintain my heart rate zone.  However I can't seem to find a place to set my heart rate (max or threshold) so that the app knows what the target should be.  Am I missing a setting somewhere - if so can someone kindly point me to it?

**Comment by cagnulein** [comment]
this one ![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/b5d2b2cd-01e5-4fb6-82eb-9b2763b3f314)

there is also a tile to change the zone at runtime and also you can programming this in a xml file so you can create a full workout based on HR

let me know if you have any other questions

**Reply by Lunchtime0614** [reply]
Yeah I found that and set it to '2' for zone 2.  But where/how does it know that my zone 2 is 132 (80% of 165)...compared to someone else who might have a max HR of 183...their zone 2 is different from mine.  Where do I set the value of 165 so it knows how to calculate my zones?

**Reply by cagnulein** [reply]
here ![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/6a84feec-213c-4986-9a89-4e84f750c88a)

**Reply by Lunchtime0614** [reply]
Is that zone % or the actual heart rate zone value?  I thought that was % as zone 2 should be 80% of threshold hr.

**Reply by Lunchtime0614** [reply]
Oh - the HR Max Override is what your referring to...got it

**Reply by cagnulein** [reply]
yep! let me know if you have any other questions

**Reply by Lunchtime0614** [reply]
Isn't there 5 HR zones?  In the zone options it's only going to zone 4

Polar shows as 5
![Screenshot from 2023-12-27 12-19-41](https://github.com/cagnulein/qdomyos-zwift/assets/68966864/b9c967e4-2ce5-44f2-bd88-39c95374e426)

**Comment by cagnulein** [comment]
4 zone borders means 5 zones :)

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mer 27 dic 2023 alle 18:20 Lunchtime0614 ***@***.***>
ha scritto:

**Reply by Lunchtime0614** [reply]
So if I change your image to 60, 70, 80, 90 (z1, z2, z3, z4) that would be the same as the polar chart?

0-60% is zone 1
61-70 z2
71-80 z3
81-90 z4
90+ z5

do I understand this correctly?

**Reply by cagnulein** [reply, marked as answer]
exactly!

**Comment by Lunchtime0614** [comment]
One last question (I think) on this - the settings for the HR zone training are within the "Training Program Options" section of the settings pages.  Does this mean for this to work I must "Load a Training Program" for it to maintain a HR Zone?  If so how does that work if the training program calls for something like 300 watts which would be well past HR Zone 2.

**Reply by cagnulein** [reply]
nope you can just do a free ride/run using this setting and I will mantain the HR without loading any training program.
Instead if you load a training program, this setting will try to keep the HR, but when a training program will call a new wattage the resistance will increase and the the HR PID function will try to lower it if you are doing too much effort

**Comment by Lunchtime0614** [comment]
Coming back to this as I tried a zone 2 target HR ride this past weekend but I couldn't get it to work.  My trainer is defined in the 'bike options' (using the beta app)...and my trainer also passes my HR.  QZ shows my power and HR on the free ride screen just fine (peloton rides work great).  But I had a custom HR Max override set from 132 to 147, I had PID zone option disabled.

When I started the ride everything on the display looked fine, however the resistance was not increased every 45 seconds as I thought to get me to my target HR metrics.  After about 3-4 min of no changes I gave up and did the ride manually.  

Is there some setting I'm not configuring properly or is there a warmup time where it wont adjust the resistance?

**Reply by cagnulein** [reply]
@Lunchtime0614 upload a debug log about this and i will check it https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
